pygsti/protocols/protocol.py

```diff
@@ -1268,7 +1313,7 @@ def from_edesign(cls, edesign, name):
             raise ValueError("Cannot convert a %s to a %s!" % (str(type(edesign)), str(cls)))
 
     def __init__(self, sub_designs, all_circuits=None, qubit_labels=None, sub_design_dirs=None,
-                 interleave=False):
+                 interleave=False, skip_writing_all_circuits=False):
         """
```

Comment:
I think the general idea of having the option to skip writing `all_circuits` is alright, but having this be something that is specified as an attribute of a class instance during the constructor stage feels suboptimal. This feels like something that would fit better as an optional flag passed into the `write` method. Of course, doing so would mean you'd need to implement an overriding version of `write` specific to `CombinedExperimentDesign`. Then again it looks like this is the tactic that was taken for the `FreeformDesign`, so maybe that isn't so bad?

Reply:
I actually tried that first, but it got a little nasty because a) it was doing a lot of manipulation of auxfile_types on the fly, and b) I really want to trigger that option as part of ProtocolData, IBMQExperiment, etc and that would mean plumbing up an option that was only used for CombinedExperimentDesign through everything that could possibly hold an edesign... Any suggestions on avoiding that plumbing are welcome, the class member was my first pass at it but I agree it's pretty awkward.

Reply:
Actually I'm not sure why I didn't think of this, but I can just check whether all_circuits_needing_data matches the set of all subdesigns... Duh. (Idea came from your assert comment below.)